### PR TITLE
Add the init and destroy method in the XATransactionManger

### PR DESCRIPTION
**Pull request by zhfeng**
Fixes https://github.com/sharding-sphere/sharding-sphere/issues/1357

### CODE REVIEW

File: sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/AbstractXATransactionManager.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.UserTransaction;
+import java.sql.SQLException;
+
+@Slf4j
+public abstract class AbstractXATransactionManager implements XATransactionManager {
```
Comment (cherrylzhao) [inline sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/AbstractXATransactionManager.java]: maybe AbstractXAtransactionManager only be used by AtomikosTransactionManager,
it think narayaTransactionManager was implemented in other repo and don't depend on SS,
so it seems that we should do init directly in AtomikosTransactionManager, and I think init() method can be removed.

File: sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/XATransactionManager.java [hunk 2]
```diff
@@ -30,6 +30,17 @@
  * @author zhangliang
  */
 public interface XATransactionManager extends ShardingTransactionManager<XATransactionEvent> {
+
+    /**
+     * initialize the transaction manager
+     */
+    void init() throws Exception;
```
Comment (cherrylzhao) [inline sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/XATransactionManager.java]: I think we can remove this init() method, and do initialize in constructor of AtomikosTransactionManager directly

File: sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/AbstractXATransactionManager.java (hunk 1 above)
Comment (zhfeng) [reply sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/AbstractXATransactionManager.java]: OK, it looks good to remove this abstract class.

File: sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/XATransactionManager.java (hunk 2 above)
Comment (zhfeng) [reply sharding-transaction/src/main/java/io/shardingsphere/transaction/manager/xa/XATransactionManager.java]: that makes sense.

**Comment by cherrylzhao** [comment]
@terrymanu please merge this PR to dev-3.1.0

**Pull request merged**